api/src/main/java/org/apache/iceberg/io/CloseableIterable.java

```diff
@@ -117,6 +118,14 @@ public O next() {
     };
   }
 
+  static <E> CloseableIterable<E> combine(Iterable<? extends Iterable<E>> iterable,
+                                          ExecutorService executorService,
+                                          int parallelism) {
+    Preconditions.checkNotNull(executorService, "ExecutorService shouldn't be null");
+    Preconditions.checkArgument(parallelism > 0, "Parallelism must be > 0");
```

Comment:
Here as well, I think the error message should follow our conventions to state the problem, give context, and then show the bad value: "Invalid parallelism (not positive): %s"